public/app/plugins/panel/nodeGraph/suggestions.ts

```diff
@@ -0,0 +1,32 @@
+import { VisualizationSuggestionsBuilder, VisualizationSuggestionScore } from '@grafana/data';
+import { SuggestionName } from 'app/types/suggestions';
+
+export class NodeGraphSuggestionsSupplier {
+  getListWithDefaults(builder: VisualizationSuggestionsBuilder) {
+    return builder.getListAppender<{}, {}>({
+      name: SuggestionName.NodeGraph,
+      pluginId: 'nodeGraph',
+    });
+  }
+
+  getSuggestionsForData(builder: VisualizationSuggestionsBuilder) {
+    if (!builder.data) {
+      return;
+    }
+
+    const dataFrames = builder.data.series.filter(
+      (df) => df.meta && df.meta.preferredVisualisationType === 'nodeGraph'
+    );
+    // one frame for nodes, one frame for edges
+    if (dataFrames.length !== 2) {
+      return;
+    }
+
+    if (builder.dataSummary.preferredVisualisationType === 'nodeGraph') {
```

Comment:
This is a bit odd. We are checking the actual data frames and then check this summary again, why is that? It seems it can also be buggy, if there are 2 node graph dataFrames but there is also third one that isnt nodegraph this will pass depending on what order those dataFrames are passed in (based on the docs of the dataSummary)